skimage/draw/_random_shapes.py

```diff
@@ -200,13 +205,34 @@ def _generate_random_colors(num_colors, num_channels, intensity_range, random):
         each channel are drawn from the corresponding `intensity_range`.
 
     """
+
     if num_channels == 1:
-        intensity_range = (intensity_range, )
+        intensity_range_tpl = (intensity_range, )
     elif len(intensity_range) == 1:
-        intensity_range = intensity_range * num_channels
+        intensity_range_tpl = intensity_range * num_channels
+    else:
+        intensity_range_tpl = intensity_range
```

Comment:
Is there really a need in changing this block?

Reply:
I wanted to keep the original intensity_range value for using in the recursive call in line 230. But I see that this approach is confusing, makes the code less readable. Indeed this can be done outside the block in a more simple way.